### DISCUSSION TITLE
ServerCertificateCustomValidationCallback

### DISCUSSION THREAD

**Discussion by bbday** [cefsharp/CefSharp#3855] [General]
Is possible get current X509Certificate while loading (OnBeforeResourceLoad) request?
My scope is detect when somebody use reverse proxy https decription (like fiddler) with X509 self signed cert, then i need to detect what is current cert.

I tried this it return right at first start then it dont "update" :
```
protected override CefReturnValue OnBeforeResourceLoad(IWebBrowser chromiumWebBrowser, IBrowser browser, IFrame frame, IRequest request, IRequestCallback callback)
        {
               var cert = chromiumWebBrowser.GetVisibleNavigationEntryAsync().Result.SslStatus;
               Console.WriteLine(cert.X509Certificate?.SubjectName);
```
but dont retunr current cert for unknow reason

**Comment by amaitland** [comment]
> Is possible get current X509Certificate while loading (OnBeforeResourceLoad) request?

OnBeforeResourceLoad is called before a request is made.

CEF doesn't support obtaining the certificate for an individual resource load. 

You can call GetVisibleNavigationEntryAsync when the page has finished loading to obtain the certificate.

**Reply by amaitland** [reply]
CEF feature requests can be made at https://magpcss.org/ceforum/viewforum.php?f=7

You can probably hack something together as a workaround, make a request for the main frame using .Net http client, check the certificate then execute the callback to continue/cancel the request.

**Reply by amaitland** [reply]
Discission moved to https://magpcss.org/ceforum/viewtopic.php?f=7&t=18703
